yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -598,8 +598,12 @@ private[spark] class Client(
     ).foreach { case (flist, resType, addToClasspath) =>
       flist.foreach { file =>
         val (_, localizedPath) = distribute(file, resType = resType)
-        if (addToClasspath && localizedPath != null) {
-          cachedSecondaryJarLinks += localizedPath
+        if (addToClasspath) {
+          if (localizedPath != null) {
+             cachedSecondaryJarLinks += localizedPath
+          }
+        } else {
+          require(localizedPath !=null)
```

Comment:
Lets change the error to illegal argument exception. 
Also lets comment this to indicate jars are ok due to spark 2.0 jar install, everything else shouldn't have multiple of same jar/file/archive.